helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -1165,7 +1164,11 @@ public void run() {
               + _processorName);
       while (!isInterrupted()) {
         try {
-          handleEvent(_eventBlockingQueue.take(), _cache);
+          ClusterEvent newClusterEvent = _eventBlockingQueue.take();
+          // rename the current processor's thread name to include event id
+          String currentThreadName = this.getName().split("_")[0];
+          this.setName(currentThreadName + "_" + newClusterEvent.getEventId());
```

Comment:
Can we use "()" to wrap the event? I feel it's easier to identify which part is the default thread name, and which part is the event name.

Reply:
Done.